src/Microsoft.Tye.Core/ConfigModel/ConfigApplication.cs

```diff
@@ -28,5 +33,139 @@ public class ConfigApplication
         public List<ConfigService> Services { get; set; } = new List<ConfigService>();
 
         public List<ConfigIngress> Ingress { get; set; } = new List<ConfigIngress>();
+
+        public void Validate()
+        {
+            var config = this;
+
+            var context = new ValidationContext(config);
+            var results = new List<ValidationResult>();
+            
+            if (!Validator.TryValidateObject(config, context, results, validateAllProperties: true))
+            {
+                throw new TyeYamlException(
+                    "Configuration validation failed." + Environment.NewLine +
+                    string.Join(Environment.NewLine, results.Select(r => r.ErrorMessage)));
+            }
+
+            foreach (var extension in config.Extensions)
+            {
+                if (!extension.TryGetValue("name", out var name) || string.IsNullOrWhiteSpace(name as string))
+                {
+                    throw new TyeYamlException(CoreStrings.ExtensionMustProvideAName);
```

Comment:
oh no resources. lol

Do you think there's a really important reason we use resources?

Reply:
I like them for having consistent strings and sharing strings between src and test.

What other mechanism would you prefer?

Reply:
I hadn't thought about sharing with tests. Makes good sense 👍